### ISSUE TITLE
How should a user specify the number of locales to use for multilocale interop libraries?

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#12854]
I thought I had made an issue for this already, but I'm not finding it right now.

Anyways, when the user creates a distributed Chapel library using --library, they
probably intend for it to run with more than just one locale.

Should the number of locales be determined by the person that created the library?
Or the person that links against the library?

In the former, we could hard-code the number of locales to use into the library
for all time.  In the latter, we could do something like provide an alternative to
`chpl_library_init` specifically for multilocale libraries.

**Comment by bradcray** [comment]
> Should the number of locales be determined by the person that created the library?
> Or the person that links against the library?

I think it's clearly the person who initializes (launches / runs) the library, similar to typical program executions.  The person who writes a library has no idea what size machine a client might want/need to run it on.

**Comment by gbtitus** [comment]
> I think it's clearly the person who initializes (launches / runs) the library, ...

Agreed.  Neither the author of the library nor someone who links with it necessarily knows the scale of data the program will be processing, and thus the appropriate locale count.
